### ISSUE TITLE
Compose debugging does not start when app path contains special characters (like umlauts)

### ISSUE THREAD

**Issue by codekoenig** [microsoft/DockerTools#338]
I use compose in Visual Studio to start & debug my app. Unfortunately, while the build and compose process seems to succeed without error, debugging was simply not starting without any errors.

After a lot of trial & error I tracked down the issue: my last name contains an Umlaut (ö) and my Windows user profile therefor has an Umlaut in it (``..\users\bernhardkönig\``).

Here is my ``docker-compose.yml`` for my small demo app:

```yaml
version: '3.4'

services:
  webapplication1:
    image: ${DOCKER_REGISTRY-}webapplication1
    build:
      context: .
      dockerfile: WebApplication1/Dockerfile
```

Here is the generated ``docker-compose.vs.debug.g.yml`` that will contain full paths with the Umlauts in it:

```yaml
version: '3.4'

services:
  webapplication1:
    image: webapplication1:dev
    container_name: WebApplication1_1
    build:
      target: base
      labels:
        com.microsoft.created-by: "visual-studio"
        com.microsoft.visual-studio.project-name: "WebApplication1"
    environment:
      - DOTNET_USE_POLLING_FILE_WATCHER=1
      - ASPNETCORE_LOGGING__CONSOLE__DISABLECOLORS=true
      - NUGET_FALLBACK_PACKAGES=/root/.nuget/fallbackpackages
    volumes:
      - C:\Users\bernhardkönig\source\repos\WebApplication1\WebApplication1:/app
      - C:\Users\bernhardkönig\source\repos\WebApplication1:/src
      - C:\Users\bernhardkönig\vsdbg\vs2017u5:/remote_debugger:rw
      - C:\Users\bernhardkönig\.nuget\packages\:/root/.nuget/packages:ro
      - C:\Program Files (x86)\Microsoft Visual Studio\Shared\NuGetPackages:/root/.nuget/fallbackpackages:ro
      - ${APPDATA}/ASP.NET/Https:/root/.aspnet/https:ro
      - ${APPDATA}/Microsoft/UserSecrets:/root/.microsoft/usersecrets:ro

    entrypoint: tail -f /dev/null
    labels:
      com.microsoft.visualstudio.debuggee.program: "dotnet"
      com.microsoft.visualstudio.debuggee.arguments: " --additionalProbingPath /root/.nuget/packages --additionalProbingPath /root/.nuget/fallbackpackages  \"/app/bin/Release/net6.0/WebApplication1.dll\""
      com.microsoft.visualstudio.debuggee.workingdirectory: "/app"
      com.microsoft.visualstudio.debuggee.killprogram: "/bin/sh -c \"if PID=$$(pidof dotnet); then kill $$PID; fi\""
    tty: true
```

To reproduce:

* With VS2022 create a new ASP.NET MVC Core Web Project in a folder named ``c:\umlautissue\``
* Add Orchestration Support / Docker Compose / Linux
* Run the demo project -everything will run smoothly and debugging is starting
* Stop the execution
* Clean the solution
* Close Visual Studio
* Make sure all containers and images are gone in Docker - if not, manually remove them
* Rename the base folder ``c:\umlautissue`` so that it contains an umlaut, replace the u with ü for example: ``c:\ümlautissue`` 
* Open the Solution from this now renamed folder
* Run the solution with Docker Compose selected
* Notice that compose runs, container gets built and deployed to Docker
* **But debugging does not start without any errors or warings**

Due to no errors or warnings or logs I could find, I have no idea where the issue is coming from. But this clue might help: a folder is created for a file named ``CanonicalServiceNames`` that clearly has encoding issues (my folder name of choice was ``tmpsöurce``):

![image](https://user-images.githubusercontent.com/18646713/166344098-f29b1c73-b9e3-44a5-871d-af252c70c22e.png)

So it might be related to this file that can't be found at a later point due to the ill encoded path name. But just a guess from my side.

**Comment by dbreshears** [comment]
Thanks for the repro.  Will need to investigate this to see why we are not maintaining the correct encoding

**Comment by dbreshears** [comment]
@codekoenig .  This is fixed and should go out with 17.3 Preview 2 release.

**Comment by dbreshears** [comment]
Should be fixed now
